src/core/src/bootstrap/Bootstrapper.py

```diff
@@ -142,36 +143,46 @@ def basic_environment_health_check(self):
+                if attempts == Constants.MAX_CHECK_SUDO_RETRY_COUNT:
+                    self.composite_logger.log_debug("Sudo Check Failed, reached [MaxAttempts={0}]".format(str(attempts)))
+                    if raise_if_not_sudo:
+                        raise
+
+                self.composite_logger.log_error("Retrying sudo status check after a delay of [ElapsedTimeInSeconds={0}][Attempts={1}]".format(Constants.MAX_CHECK_SUDO_INTERVAL_IN_SEC, str(attempts)))
+                time.sleep(Constants.MAX_CHECK_SUDO_INTERVAL_IN_SEC)
+
     def check_sudo_status(self, raise_if_not_sudo=True):
         """ Checks if we can invoke sudo successfully. """
-        try:
-            self.composite_logger.log("Performing sudo status check... This should complete within 10 seconds.")
-            return_code, output = self.env_layer.run_command_output("timeout 10 sudo id && echo True || echo False", False, False)
-            # output should look like either this (bad):
-            #   [sudo] password for username:
-            #   False
-            # or this (good):
-            #   uid=0(root) gid=0(root) groups=0(root)
-            #   True
-
-            output_lines = output.splitlines()
-            if len(output_lines) < 2:
-                raise Exception("Unexpected sudo check result. Output: " + " ".join(output.split("\n")))
-
-            if output_lines[1] == "True":
-                return True
-            elif output_lines[1] == "False":
-                if raise_if_not_sudo:
-                    raise Exception("Unable to invoke sudo successfully. Output: " + " ".join(output.split("\n")))
-                return False
-            else:
-                raise Exception("Unexpected sudo check result. Output: " + " ".join(output.split("\n")))
-        except Exception as exception:
-            self.composite_logger.log_error("Sudo status check failed. Please ensure the computer is configured correctly for sudo invocation. " +
-                                            "Exception details: " + str(exception))
+        return_code, output = self.env_layer.run_command_output("timeout 10 sudo id && echo True || echo False", False, False)
+        # output should look like either this (bad):
+        #   [sudo] password for username:
+        #   False
+        # or this (good):
+        #   uid=0(root) gid=0(root) groups=0(root)
+        #   True
+        output_lines = output.splitlines()
+        if len(output_lines) < 2:
+            raise Exception("Unexpected sudo check result. Output: " + " ".join(output.split("\n")))
+
+        if output_lines[1] == "True":
+            return True
+        elif output_lines[1] == "False":
             if raise_if_not_sudo:
-                raise
-
+                raise Exception("Unable to invoke sudo successfully. Output: " + " ".join(output.split("\n")))
+            return False
+        else:
+            raise Exception("Unexpected sudo check result. Output: " + " ".join(output.split("\n")))
```

Comment:
I'm not seeing a good reason to change any of this - the highlighted note:

![image](https://github.com/user-attachments/assets/bfbad255-3b4a-461c-a9a9-28151c90e095)